### ISSUE TITLE
Startinsert only specific patterns

### ISSUE THREAD

**Issue by yutkat** [voldikss/vim-floaterm#295]
### Before reporting

I want to do startinsert only for the following patterns. Because I'm using Floaterm to run tests, etc.

``` vim
let g:floaterm_autoinsert = v:false
autocmd TermOpen,TermEnter term://*/zsh startinsert
```

But I can't do that because the following line will force the mode back to normal.
https://github.com/voldikss/vim-floaterm/blob/b7747704c4df716efd21f331b4d94336c490a5ad/autoload/floaterm/util.vim#L60

I would like you to remove that line or add an option to skip it.

### Steps to reproduce

Using the minimal vimrc

```vim
set nocompatible
set hidden
set termguicolors
set runtimepath^=/path/to/vim-floaterm
filetype plugin indent on
syntax on
let g:floaterm_autoinsert = v:false
autocmd TermOpen,TermEnter term://*/zsh startinsert
" autocmd FloatermOpen term://*/zsh startinsert " -> overwritten
```

Steps to reproduce:
1. `:FloatermNew`
2. Check the mode

### Actual behaviour

Return to normal mode when opening zsh on floaterm

### Expected behaviour

I want to do startinsert only when the execution is zsh.

**Comment by adigitoleo** [comment]
I propose that this option should be removed completely. Startinsert can be controlled by the users with autocommands, there is no need for us to have this.

**Comment by voldikss** [comment]
> I propose that this option should be removed completely. Startinsert can be controlled by the users with autocommands, there is no need for us to have this.

RIGHT!!! I feel that the plugin has exposed so much interfaces/options which users can implement via easy vimscript. These redunant options increases the complexity of the plugin and I'm considering to remove them.

**Comment by voldikss** [comment]
`g:floaterm_autoinsert` now accepts `'never'`, and `v:false` is treated as `'never'` for backward compatibility. With this setting the plugin never enters Terminal-mode on its own and never forces the mode back to normal, so your approach now works:

```vim
let g:floaterm_autoinsert = 'never'  " v:false is equivalent now
autocmd TermOpen,TermEnter term://*/zsh startinsert
```

Floaterms running zsh enter Terminal-mode via your autocmd, everything else stays in normal mode.

Closing — feel free to reopen if this doesn't cover your use case.
